src/include/duckdb/main/secret/secret_storage.hpp

```diff
@@ -36,17 +38,19 @@ class SecretStorage {
 	};
 
 	//! Store a secret
-	virtual optional_ptr<SecretEntry> StoreSecret(CatalogTransaction transaction, unique_ptr<const BaseSecret> secret,
-	                                              OnCreateConflict on_conflict) = 0;
+	virtual optional_ptr<SecretEntry> StoreSecret(unique_ptr<const BaseSecret> secret, OnCreateConflict on_conflict,
+	                                              optional_ptr<CatalogTransaction> transaction = nullptr) = 0;
 	//! Get all secrets
-	virtual vector<reference<SecretEntry>> AllSecrets(CatalogTransaction transaction) = 0;
+	virtual vector<reference<SecretEntry>> AllSecrets(optional_ptr<CatalogTransaction> transaction = nullptr) = 0;
```

Comment:
Can we make SecretEntry copyable and return either an owning ptr (i.e `unique_ptr<SecretEntry>`) or a copy (i.e. `SecretEntry`) in `AllSecrets`, `GetSecretByName`, and `LookupSecret` (`SecretMatch` has an non-owning `optional_ptr` to `SecretEntry` and should preferably be `unique_ptr`)? This way we don’t lose the secret object by the time we return the secret-related output from our custom secret storage to the duck secret manager since our secrets are not tied to a Catalog and will need to be re-created on the server side each time upon lookup.

Proposed changes:
1. SecretMatch:
```
struct SecretMatch {
    ...
    unique_ptr<SecretEntry> secret_entry;
    int64_t score;
};
```
2. SecretStorage read APIs:
```
virtual vector<unique_ptr<SecretEntry>> AllSecrets(...)
virtual unique_ptr<SecretEntry> GetSecretByName(...)
```